python-avd/pyavd/_eos_designs/eos_designs_facts/short_esi.py

```diff
@@ -29,6 +29,10 @@ def _short_esi(self: EosDesignsFacts) -> str:
         """
         short_esi = get(self.shared_utils.switch_data_combined, "short_esi")
         if short_esi == "auto":
+            if self.shared_utils.mlag_role == "secondary":
+                # On the MLAG Secondary use short-esi from MLAG primary
+                if (peer_short_esi := self.shared_utils.mlag_peer_facts._short_esi) is not None:
+                    return peer_short_esi
```

Comment:
This logic should be above everything else so above line 30, otherwise it will only kick in if short_esi is auto.

Reply:
Done!